appcelerator/titanium_mobile
# fix(ios): missing Ti.App.arguments and getArguments() APIs

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28507

**Summary:**
- New property `Ti.App.arguments` wrongly returns undefined as of 10.0.0.
- Existing `Ti.App.getArguments()` method was wrongly removed as of 10.0.0.
  * We documented it as deprecated, not removed, for 10.0.0.
- Since above APIs were missing, it made it impossible to fetch app launch arguments which is a major regression.

---
**Test:**
1. Set up a project's "tiapp.xml" and "app.js" with the below.
2. Build and run on iOS.
3. Verify you see the follow text in the app on startup.
```
[INFO] Ti.App.arguments:
[INFO] {}
```
4. Force quit the app.
5. Tap and hold on the app icon and select "Static Shortcut 1" from the menu.
6. Verify you see the following text on app startup.
```
[INFO] Ti.App.arguments:
[INFO] {
[INFO]     "UIApplicationLaunchOptionsShortcutItemKey": {
[INFO]         "type": "static_shortcut1",
[INFO]         "title": "Static Shortcut 1",
[INFO]         "subtitle": "My subtitle"
[INFO]     }
[INFO] }
```

---
tiapp.xml
```xml
<ti:app>
	<ios>
		<plist>
			<dict>
				<key>UIApplicationShortcutItems</key>
				<array>
					<dict>
						<key>UIApplicationShortcutItemIconType</key>
						<string>UIApplicationShortcutIconTypeSearch</string>
						<key>UIApplicationShortcutItemTitle</key>
						<string>Static Shortcut 1</string>
						<key>UIApplicationShortcutItemSubtitle</key>
						<string>My subtitle</string>
						<key>UIApplicationShortcutItemType</key>
						<string>static_shortcut1</string>
					</dict>
				</array>
			</dict>
		</plist>
	</ios>
</ti:app>
```

app.js
```javascript
const window = Ti.UI.createWindow({
	extendSafeArea: false,
	layout: "vertical",
});
window.add(Ti.UI.createLabel({
	text: `Ti.App.arguments:\n${JSON.stringify(Ti.App.arguments, null, 4)}`,
	top: "30dp",
}));
window.open();
```
